picard/file_dilaog.py

```diff
@@ -0,0 +1,274 @@
+                self.show_files(False)
+
+    def show_files(self,hidden):
+        self.hidden_items = 0
+        self.files_to_show()
+        if hidden == False :
+            self.table.setRowCount(len(self.dirs)+len(self.files)-self.hidden_items)
+        else:
+            self.table.setRowCount(len(self.dirs)+len(self.files))
+        start,pos,j=0,0,0
+        while start < len(self.dirs):
+            if hidden == False :
+                if Is_hidden(self.dirs[start]) == True:
+                    start=start+1
+                    continue
+            self.table.setItem(pos,0,QTableWidgetItem("      "+self.dirs[start]))
+            self.table.setItem(pos,1,QTableWidgetItem(self.dirs_size[start]))
+            self.table.setItem(pos,2,QTableWidgetItem(self.dirs_modi_time[start]))
+            self.table.setCellWidget(pos, 0, ImgWidget1("../resources/img-src/folder24_24.png"))
+            pos,start=pos+1,start+1
+
+        while j < len(self.files):
+            if hidden == False :
+                if Is_hidden(self.files[j]) == True:
+                    j=j+1
+                    continue
+            self.table.setItem(pos,0,QTableWidgetItem("      "+self.files[j]))
+            self.table.setItem(pos,1,QTableWidgetItem(self.files_size[j]))
+            self.table.setItem(pos,2,QTableWidgetItem(self.files_modi_time[j]))
+            self.table.setCellWidget(pos, 0, ImgWidget1("../resources/img-src/music24_24.png"))
+            pos,j=pos+1,j+1
+
+    def search_update(self):
+        print(self.lineedit.text())
+        self.lineedit.setText("")
+
+    def show_hidden(self,cb):
+        if cb.isChecked() == True:
+            self.show_files(True)
+        else:
+            self.show_files(False)
+
+    def files_to_show(self):
+        self.files = []
+        self.files_size = []
+        self.files_modi_time = []
+        self.dirs = []
+        self.dirs_size = []
+        self.dirs_modi_time = []
+        def converter(num):
+            for x in ['bytes', 'KB', 'MB', 'GB', 'TB']:
+                if num < 1024.0:
+                    return "%3.1f %s" % (num, x)
+                num /= 1024.0
+
+        for i in os.listdir(self.path):
+            file_info = os.stat(os.path.join(self.path,i))
+            if Is_hidden(i):
+                if os.path.isfile(os.path.join(self.path,i)):
+                    for j in self.formats_lower:
+                        if i.endswith('.'+j):
+                            self.hidden_items=self.hidden_items+1
+                else:
+                    self.hidden_items=self.hidden_items+1
+            date_= str((datetime.fromtimestamp(file_info.st_mtime)).date())
+            if os.path.isdir(os.path.join(self.path,i)):
+                self.dirs.append(i)
+                self.dirs_size.append("NA")
+                self.dirs_modi_time.append(date_)
+            else:
+                for j in self.formats_lower:
+                    if i.endswith('.'+j) == True:
```

Comment:
What's `j` here ?
Also, in Python, we don't usually test against False / True: `if i.endswith('.' + j):`